test/powershell/Modules/Microsoft.PowerShell.Utility/Remove-Alias.Tests.ps1

```diff
@@ -0,0 +1,58 @@
+Describe "Remove-Alias" -Tags "CI" {
+    It "Remove-Alias should remove a non-readonly alias"{
+        {
+            Set-Alias -Name "tral" -Value "Remove-Alias" -ErrorAction Stop
+            Remove-Alias -Name "tral" -ErrorAction Stop
+            Get-Alias -Name "tral" -ErrorAction Stop
+        } | ShouldBeErrorId 'ItemNotFoundException,Microsoft.PowerShell.Commands.GetAliasCommand'
+    }
+
+    It "Remove-Alias should throw on a readonly alias"{
+        {
+            Set-Alias -Name "tral" -Value "Remove-Alias" -Option ReadOnly -ErrorAction Stop
+            Remove-Alias -Name "tral" -ErrorAction Stop
+        } | ShouldBeErrorId 'AliasNotRemovable,Microsoft.PowerShell.Commands.RemoveAliasCommand'
+    }
+
+    It "Remove-Alias should remove a non-readonly alias with force"{
+        {
+            Set-Alias -Name "tral" -Value "Remove-Alias" -ErrorAction Stop
+            Remove-Alias -Name "tral" -Force -ErrorAction Stop
+            Get-Alias -Name "tral" -ErrorAction Stop
+        } | ShouldBeErrorId 'ItemNotFoundException,Microsoft.PowerShell.Commands.GetAliasCommand'
+    }
+
+    It "Remove-Alias should remove a readonly alias with force"{
+        {
+            Set-Alias -Name "tral" -Value "Remove-Alias" -Option ReadOnly -ErrorAction Stop
+            Remove-Alias -Name "tral" -Force -ErrorAction Stop
+            Get-Alias -Name "tral" -ErrorAction Stop
+        } | ShouldBeErrorId 'ItemNotFoundException,Microsoft.PowerShell.Commands.GetAliasCommand'
+    }
+
+    It "Remove-Alias should throw if alias does not exist"{
```

Comment:
i assume that this is a non-terminating error - should there be a test for that? Something like:
```
Remove-Alias -Name "tral" -ErrorVariable RemoveAliasError 2>$null
$RemoveAliasError.FullyQualifiedErrorId | ShouldBe "ItemNotFoundException,Microsoft.PowerShell.Commands.RemoveAliasCommand"
```

Reply:
We have tons of such "incorrect" tests. I think we should refactor them all in time or after resolving #4781